cartography/intel/gcp/pubsub.py

```diff
@@ -0,0 +1,232 @@
+            page_start = (common_job_parameters.get('pagination', {}).get('pubsub', None)[
+                          'pageNo'] - 1) * common_job_parameters.get('pagination', {}).get('pubsub', None)['pageSize']
+            page_end = page_start + common_job_parameters.get('pagination', {}).get('pubsub', None)['pageSize']
+            if page_end > len(topics) or page_end == len(topics):
+                topics = topics[page_start:]
+            else:
+                has_next_page = True
+                topics = topics[page_start:page_end]
+                common_job_parameters['pagination']['pubsub']['hasNextPage'] = has_next_page
+
+        return topics
+    except HttpError as e:
+        err = json.loads(e.content.decode('utf-8'))['error']
+        if err.get('status', '') == 'PERMISSION_DENIED' or err.get('message', '') == 'Forbidden':
+            logger.warning(
+                (
+                    "Could not retrieve pubsub topics on project %s due to permissions issues. Code: %s, Message: %s"
+                ), project_id, err['code'], err['message'],
+            )
+            return []
+        else:
+            raise
+
+
+@timeit
+def load_pubsub_topics(session: neo4j.Session, data_list: List[Dict], project_id: str, update_tag: int) -> None:
+    session.write_transaction(load_pubsub_topics_tx, data_list, project_id, update_tag)
+
+
+@timeit
+def load_pubsub_topics_tx(
+    tx: neo4j.Transaction, data: List[Dict],
+    project_id: str, gcp_update_tag: int,
+) -> None:
+
+    query = """
+    UNWIND {Records} as record
+    MERGE (topic:GCPPubsubTopic{id:record.id})
+    ON CREATE SET
+        topic.firstseen = timestamp()
+    SET
+        topic.lastupdated = {gcp_update_tag},
+        topic.region = record.region,
+        topic.name = record.topic_name,
+        topic.kms_key_name = record.kmsKeyName,
+        topic.satisfies_pzs = record.satisfiesPzs,
+        topic.message_retention_duration = record.messageRetentionDuration
+    WITH topic
+    MATCH (owner:GCPProject{id:{ProjectId}})
+    MERGE (owner)-[r:RESOURCE]->(topic)
+    ON CREATE SET
+        r.firstseen = timestamp()
+    SET r.lastupdated = {gcp_update_tag}
+    """
+    tx.run(
+        query,
+        Records=data,
+        ProjectId=project_id,
+        gcp_update_tag=gcp_update_tag,
+    )
+
+
+@timeit
+def cleanup_pubsub_topics(neo4j_session: neo4j.Session, common_job_parameters: Dict) -> None:
+    run_cleanup_job('gcp_pubsub_topics_cleanup.json', neo4j_session, common_job_parameters)
+
+
+@timeit
+def sync(
+    neo4j_session: neo4j.Session, pubsub: Resource, project_id: str, gcp_update_tag: int,
+    common_job_parameters: Dict, regions: List,
+) -> None:
+
+    tic = time.perf_counter()
+    logger.info("Syncing Pubsub for project '%s', at %s.", project_id, tic)
+
+    subscriptions = get_pubsub_subscriptions(pubsub, project_id, regions, common_job_parameters)
```

Comment:
we should establish relationships between topic & subscriptions. because, subscriptions can't be created on their own. they always attach with topics.